File: basel.py
import math as m

def s(x):
	s = sum(i for i in range(1, x) if x % i == 0)
	return s

def σ(x):
	σ = sum(i for i in range(1, x+1) if x % i == 0)
	return σ

def basel(n):
	S = 0
	for i in range(1, n+1):
		S += σ(i)/i
	print("[{:d}]	σ(n)/n = {:.2f}		S/n = {:.16f}	S > {:d}".format(i, σ(i)/i, S/i, int(S)))
	print("--- Final ---")

def basel_simulation(n):
	basel = m.pi**2 / 6
	print("basel = {:.10f}".format(basel))
	print("S ~ {:d}".format(int(basel * n)))

n = 10000

basel(n)

basel_simulation(n)

# Imprimindo números 	: 0m15,675s
# Sem imprimir 			: 0m3,487s

File: basel.c
#include <stdio.h>
#include <math.h>

int s(int x) {
    int sum = 0;
    for (int i = 1; i < x; i++) {
        if (x % i == 0) {
            sum += i;
        }
    }
    return sum;
}

int sigma(int x) {
    int sum = 0;
    for (int i = 1; i <= x; i++) {
        if (x % i == 0) {
            sum += i;
        }
    }
    return sum;
}

int main() {
    int n = 500000;
    double S = 0.0;
    int i;
    for (i = 1; i < n; i++) {
        printf("[%d]\tσ(n)/n = %.2f\t\tS/n = %.10f\tpi ~ %.8f\n", i, (double)sigma(i) / i, S / i, sqrt(6 * S / i));
        S += (double)sigma(i) / i;
    }
    
    return 0;
}

// gcc basel.c -o basel -lm